### DISCUSSION TITLE
accessible_by strategies

### DISCUSSION THREAD

**Discussion by GeorgeMarcusGG** [CanCanCommunity/cancancan#784] [Q&A]
This weekend we had an issue on production caused by `accessible_by` performance. During my investigation I discovered [SQL Strategies](https://github.com/CanCanCommunity/cancancan/blob/develop/docs/sql_strategies.md).

But why do we need these strategies? I found it weird that it joins the table with itself. What is the danger of just applying the conditions on top of the current query?

Could that be a separate strategy option? It would work in my case (currently I'm manually applying conditions to make the query faster).

**Comment by coorasse** [comment]
Can you provide an example of your situation?
